hudi-common/src/main/java/org/apache/hudi/common/table/timeline/HoodieDefaultTimeline.java

```diff
@@ -136,6 +136,14 @@ public HoodieTimeline filterPendingCompactionTimeline() {
         instants.stream().filter(s -> s.getAction().equals(HoodieTimeline.COMPACTION_ACTION) && !s.isCompleted()), details);
   }
 
+  @Override
+  public HoodieTimeline filterPendingCompactionAndReplaceTimeline() {
```

Comment:
Not required in this PR.  In general, should we have something like `filterPendingTableServicesTimeline()` so that any new async table service should not cause the same problem of staining the timeline again?  or maybe filter `!s.isCompleted()` only?

As a side note, the `COMPACTION_ACTION` only appears if it is not complete.  Once the compaction is complete, it shows as a completed `commit` instead of completed `compaction` in the timeline.

Reply:
Nice idea.
For now there are three kinds of table services including `CLUSTER`, `COMPACT` and `CLEAN` . And only commit service could stain the active timeline.
With the continuous development of HUDI, there may be more and more types of commits service. It's better to put some restrictions in case that stain the active timeline by accident.
I may spend time on this. Maybe contribute a another PR to make it happen :)